clang/lib/Lex/HeaderSearch.cpp

```diff
@@ -1801,6 +1801,33 @@ HeaderSearch::loadModuleMapFileImpl(FileEntryRef File, bool IsSystem,
   return LMM_NewlyLoaded;
 }
 
+HeaderSearch::LoadModuleMapResult
+HeaderSearch::parseModuleMapFileImpl(FileEntryRef File, bool IsSystem,
```

Comment:
There seems to be a difference between pre-parsing and parsing (pre-parse + load). This function is named as one but implements the other. Can we clarify the naming throughout this patch?

Also, the returned enum has `LMM_NewlyLoaded` enumerator, but since this function only performs pre-parsing, it should never actually load the module (i.e. materialize the `Module` instance), right?